### PR TITLE
Annotations batch job - fix duplicates & further filtering

### PR DESCRIPTION
**Pull request by JosephLavagna**
- Annotations table no longer contains duplicate names
- Added a new DateTime column to Publications table - AnnotationsSynced
- This column will be NULL if no annotations have been synced and the current DateTime when annotations synced.

- Will only pull in annotations if : 
1. Publications have been accepted by Biobank
2. AnnotationsSynced == Null
3. AnnotationsSynced > 1 month ago

- Will need to test batch job on test once databases have done migrations.

### CODE REVIEW

File: src/Publications/Publications/Services/AnnotationService.cs [hunk 1]
```diff
@@ -65,12 +64,25 @@ public async Task AddPublicationAnnotations(string publicationId, IEnumerable<An
                 }
                 else
                 {
-                    //Update existing record
-                    older.Name = newer.Name;
-                    older.Uri = newer.Uri;
-                   _ctx.Update(older);
+                    //Check if publicationAnnotation already exists 
+                    var publicationAnnotation = older.PublicationAnnotations.Where(x => x.Publication_Id == publication.Id).FirstOrDefault();
```
Comment (Flemingjp) [inline src/Publications/Publications/Services/AnnotationService.cs:68]: Can be written as

```c#
older.PublicationAnnotations.FirstOrDefault(x => x.Publication_Id == publication.Id);
```

File: src/Publications/PublicationsAzureFunctions/Testing.cs [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+            _epmcService = epmcService;
+            _fetchPublicationsService = fetchPublicationsService;
+            _fetchAnnotationsService = fetchAnnotationsService;
+        }
+
+        [FunctionName("Testing")]
+        public async Task<IActionResult> Run(
+            [HttpTrigger(AuthorizationLevel.Function, "get", "post", Route = null)] HttpRequest req,
+            ILogger log)
+        {
```
Comment (Flemingjp) [inline src/Publications/PublicationsAzureFunctions/Testing.cs:37]: Is this needed? Is it a new function or just for debugging?

File: src/Directory/Data/Migrations/202102121530124_AddAnnotationsSyncedColumnToPublications.cs [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+{
+    using System;
+    using System.Data.Entity.Migrations;
+    
+    public partial class AddAnnotationsSyncedColumnToPublications : DbMigration
+    {
+        public override void Up()
+        {
+            AddColumn("dbo.Publications", "AnnotationsSynced", c => c.DateTime());
+        }
```
Comment (Flemingjp) [inline src/Directory/Data/Migrations/202102121530124_AddAnnotationsSyncedColumnToPublications.cs:11]: Merge migrations if able to?

**Review by Flemingjp — CHANGES_REQUESTED** [review]

File: src/Publications/PublicationsAzureFunctions/Testing.cs (hunk 2 above)
Comment (JosephLavagna) [reply src/Publications/PublicationsAzureFunctions/Testing.cs:37]: @blargly pointed this out too - didnt realise it was committed - will delete now

**Review by blargly — APPROVED** [review]

**Review by blargly — APPROVED** [review]

**Review by Flemingjp — APPROVED** [review]

**Pull request merged**